toy_robot.rb

```diff
@@ -7,34 +7,47 @@ def initialize
     @x = 0
     @y = 0
     @facing = :north
+    @movements = []
   end
 
   def place(x,y,facing)
     raise InvalidPositionError.new("Invalid Position, #{x}, #{y}, #{facing}") if !(x.between?(0,4)) || !(y.between?(0,4))
     @x = x
     @y = y
     @facing = facing
+    @movements.replace(["PLACE #{@x}, #{@y}, #{@facing.to_s.upcase}"])
   end
 
   def move
```

Comment:
Hi @itinsley , I am trying to understand how far should I go to separate the code for refactor. I found it a bit hard to understand. Should I write functions separately for these:
1. Validating the new position before the 'place' command is called, the position is valid if x is between 0 - 4 and y is between 0 - 4.
2. If the position is valid, then do the 'place' command, otherwise will raise error message.
3. Validating the new position before the 'move' command is called, the position is valid if:
    
- If Robot is facing north and y is < 4
- If Robot is facing south and y is > 0
- If Robot is facing east and x is < 4
- If Robot is facing west and x is > 0

If the position is valid then the Robot will move, otherwise, it will raise error message.
Sorry, I just want to make sure.